src/java/com/sarality/app/view/list/SearchListComponentAdapter.java

```diff
@@ -0,0 +1,61 @@
+package com.sarality.app.view.list;
+
+import android.content.Context;
+import android.widget.Filter;
+import android.widget.Filterable;
+
+import com.sarality.app.common.collect.Lists;
+import com.sarality.app.view.action.ComponentActionManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Custom adapter that can filter the list and create new list
+ *
+ * @author sunayna@dothat.in sunayna
+ */
+public class SearchListComponentAdapter<T> extends ListComponentAdapter<T> implements Filterable {
+
+  private final List<T> rowValueList;
+  private final ComponentComparator comparator;
+
+  public SearchListComponentAdapter(Context context, ListRowRenderer<T> rowRenderer, List<T> rowValueList,
+                                    ComponentActionManager componentManager, ComponentComparator comparator) {
+
+    super(context, rowRenderer, rowValueList, componentManager);
+    // Make a coopy of the list that will not get manipulated when the list data is changed due to search
+    List<T> newList = Lists.of();
```

Comment:
New list will be lost as soon as the constructor completes? You may need to decide what you want to use as your original List and what you want to use as your filtered list. Not very clear what you are doing with newList here.